src/parser/monk/mistweaver/modules/talents/Upwelling.js

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import { formatNumber, formatPercentage } from 'common/format';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import Combatants from 'parser/shared/modules/Combatants';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+
+
+//const debug = false;
+
+const BASE_HOT_TIME = 8000;//ef's hot base time
+const BASE_BOLTS = 17;//18 base but we start counting at 0 so 18th on bolt count = 19th bolt
+
+
+/**
+ * Upwelling is a talent that increases essence font's hot by 4 seconds and for every second off cooldown up to 18 seconds it provides an extra bolt for the next ef cast
+ * NOTE the buff this provides for extra bolts is invisible so we get to do some ugly coding
+ * Things to Track
+ * Extra bolts from Upwelling
+ * Extra hot time from NON-upwelling bolts
+ * Total hot time from upwelling bolts
+ */
+class Upwelling extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  totalHealing = 0;
+  totalOverhealing = 0;
+  totalAbsorbs = 0;
+
+  castEF = 0;//casts of ef 
+  extraHots = 0;//number of extra hots
+  extraBolts = 0;//number of extra bolts
+  efHotHeal = 0;//healing from hots
+  efHotOverheal = 0;//overhealing from hots
+
+  boltCount = 0;
+
+  totalBolts = 0;
+
+  hotMap = new Map();//tracking on who has what and how it is
+
+  masteryTickTock = false;
+  masteryHit = 0;
+  masteryHealing = 0;
+  masteryOverhealing = 0;
+  masteryAbsorbed = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.ESSENCE_FONT_BUFF), this.efHeal);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.ESSENCE_FONT), this.efcast);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.ESSENCE_FONT_BUFF), this.applyBuff);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.ESSENCE_FONT_BUFF), this.removeBuff);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GUSTS_OF_MISTS), this.handleMastery);
+  }
+
+  efHeal(event) {
+    if(event.tick){
+      this.hotHeal(event);
+    }else{
+      this.boltHeal(event);
+    }
+  }
+
+  hotHeal(event){
+    const targetID = event.targetID;//short hand
```

Comment:
Not sure this comment is adding value, if you truly wanted shorthand here you could try,
```js
const { targetID } = event;

// or

hotHeal({ targetID}) {...}
```

Reply:
Didn't know you could do this but short hand was more or less just meaning so I don't have to keep typing `event.targetID`. :P Will do it next time though :)